src/theory/quantifiers/ce_guided_conjecture.cpp

```diff
@@ -0,0 +1,797 @@
+/*********************                                                        */
+/*! \file ce_guided_conjecture.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief implementation of class that encapsulates counterexample-guided instantiation
+ **        techniques for a single SyGuS synthesis conjecture
+ **/
+#include "theory/quantifiers/ce_guided_conjecture.h"
+
+#include "expr/datatype.h"
+#include "options/quantifiers_options.h"
+#include "smt/smt_statistics_registry.h"
+#include "theory/quantifiers/first_order_model.h"
+#include "theory/quantifiers/term_database_sygus.h"
+#include "theory/theory_engine.h"
+#include "prop/prop_engine.h"
+
+using namespace CVC4::kind;
+using namespace std;
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+
+void collectDisjuncts( Node n, std::vector< Node >& d ) {
+  if( n.getKind()==OR ){
+    for( unsigned i=0; i<n.getNumChildren(); i++ ){
+      collectDisjuncts( n[i], d );
+    }
+  }else{
+    d.push_back( n );
+  }
+}
+
+CegConjecture::CegConjecture( QuantifiersEngine * qe, context::Context* c )
```

Comment:
Is there a reason why context `c` is not used?

Reply:
At some point I was using c, now I don't need it.  I'd prefer keeping it here since it may come back later

Reply:
Would it not be better to remove it for now and if you need again later you add it? Otherwise, if you don't need it anymore it will stay there.

Reply:
I agree - I want to be fairly aggressive about pruning unused code.  It's not hard to add it back if you need it later.